web/libs/editor/src/stores/AppStore.js

```diff
@@ -798,11 +798,18 @@ export default types
       self.suggestionsRequest = requestId;
 
       self.setFlags({ awaitingSuggestions: true });
-      const response = yield request;
 
-      if (requestId === self.suggestionsRequest) {
-        self.annotationStore.selected.setSuggestions(dataParser(response));
+      try {
+        const response = yield request;
+
+        if (requestId === self.suggestionsRequest) {
+          self.annotationStore.selected.setSuggestions(dataParser(response));
+          self.setFlags({ awaitingSuggestions: false });
+        }
+      } catch(e) {
         self.setFlags({ awaitingSuggestions: false });
+        // @todo handle errors + situation when task is changed
+        return;
```

Comment:
maybe im misreading/misunderstanding the code but is this return needed? doesn't really appear to be giving us an early return. Am i missing something?
```suggestion
```